platform/config/ensureDeepObject.ts

```diff
@@ -1,29 +1,42 @@
 const separator = '.';
 
+/**
+ * Recursively traverses through the object's properties and expands ones with
+ * dot-separated names into nested objects (eg. { a.b: 'c'} -> { a: { b: 'c' }).
+ * @param {*} obj Object to traverse through
+ * @returns {Object} Same object instance with expanded properties.
```

Comment:
Not sure we should include types in these, or just rely on TypeScript and have e.g.

```
* @param obj Object to traverse through
* @returns Same object instance with expanded properties.
```

(The type isn't required in jsdoc)